test/Datadog.Trace.IntegrationTests/DiagnosticListeners/AspNetCoreMvcTestData.cs

```diff
@@ -35,82 +35,140 @@ public static class AspNetCoreMvcTestData
+            { "/I/dont/123/exist/", 404, false, "GET /i/dont/?/exist/", EmptyTags(), 1, null, null, null, null },
+            { "/Home/Error", 500, true, "GET /home/error", ConventionalParentTags(), 2, null, ConventionalChildTags(action: "error"), null, null },
+            { "/Home/UncaughtError", 500, true, "GET /home/uncaughterror", ConventionalParentTags(), 2, null, ConventionalChildTags(action: "uncaughterror"), null, null },
+            { "/Home/BadHttpRequest", 400, true, "GET /home/badhttprequest", ConventionalParentTags(), 2, null, ConventionalChildTags(action: "badhttprequest"), null, null },
+            { $"{CustomHandlerPrefix}/Home/Error", 500, true, $"GET {CustomHandlerPrefix}/home/error", ConventionalParentTags(), 2, null, ConventionalChildTags(action: "error"), null, null },
+            { $"{CustomHandlerPrefix}/Home/UncaughtError", 500, true, $"GET {CustomHandlerPrefix}/home/uncaughterror", ConventionalParentTags(), 2, null, ConventionalChildTags(action: "uncaughterror"), null, null },
+            { $"{CustomHandlerPrefix}/Home/BadHttpRequest", 500, true, $"GET {CustomHandlerPrefix}/home/badhttprequest", ConventionalParentTags(), 2, null, ConventionalChildTags(action: "badhttprequest"), null, null },
+            { $"{CustomHandlerPrefix}/throws", 500, true, $"GET {CustomHandlerPrefix}/throws", EmptyTags(), 1, null, null, null, null },
+            { $"{ExceptionPagePrefix}/Home/Error", 500, true, $"GET {ExceptionPagePrefix}/home/error", ConventionalParentTags(), 2, null, ConventionalChildTags(action: "error"), null, null },
+            { $"{ExceptionPagePrefix}/Home/BadHttpRequest", 400, true, $"GET {ExceptionPagePrefix}/home/badhttprequest", ConventionalParentTags(), 2, null, ConventionalChildTags(action: "badhttprequest"), null, null },
+            { $"{ExceptionPagePrefix}/throws", 500, true, $"GET {ExceptionPagePrefix}/throws", EmptyTags(), 1, null, null, null, null },
+            { $"{ReExecuteHandlerPrefix}/Home/Error", 500, true, $"GET {ReExecuteHandlerPrefix}/home/error", ConventionalParentTags(), 3, null, ConventionalChildTags(action: "error"), $"GET {ReExecuteHandlerPrefix}/home/index", ConventionalChildTags() },
+            { $"{ReExecuteHandlerPrefix}/Home/BadHttpRequest", 500, true, $"GET {ReExecuteHandlerPrefix}/home/badhttprequest", ConventionalParentTags(), 3, null, ConventionalChildTags(action: "badhttprequest"), $"GET {ReExecuteHandlerPrefix}/home/index", ConventionalChildTags() },
+            { $"{ReExecuteHandlerPrefix}/throws", 500, true, $"GET {ReExecuteHandlerPrefix}/throws", EmptyTags(), 2, $"GET {ReExecuteHandlerPrefix}/home/index", ConventionalChildTags(), null, null },
+            { $"{StatusCodeReExecutePrefix}/I/dont/123/exist/", 404, false, $"GET {StatusCodeReExecutePrefix}/i/dont/?/exist/", EmptyTags(), 2, $"GET {StatusCodeReExecutePrefix}/home/index", ConventionalChildTags(), null, null },
```

Comment:
I think this could really benefit from some "snapshot" style testing for the spans, but I'll tidy that up in a subsequent PR

Reply:
You may want to speak with @Kyle-Verhoog. He was prototyping a tracer-agnostic tool to record and compare trace snapshots. He showed it during one of our team demos. Cool stuff.

Reply:
:+1: we use them in Python and they've helped catch a couple of regressions already! I'm happy to help get the test agent setup in your CI if you're interested in trying it out 🙂

Reply:
Thanks @Kyle-Verhoog, we're going through some major-ish updates to the CI system at the moment, so once those have settled down I'll definitely reach out and look at getting that set up, thanks 🙂